ocaml/asmcomp/asmlink.ml

```diff
@@ -102,7 +106,12 @@ let check_consistency file_name unit crc =
 let extract_crc_interfaces () =
   Cmi_consistbl.extract !interfaces crc_interfaces
   |> List.map (fun (name, crc_with_unit) ->
-      Import_info.create name ~crc_with_unit)
+      let cu, crc =
+        match crc_with_unit with
+        | None -> None, None
+        | Some (impl, crc) -> impl, Some crc
+      in
```

Comment:
I'm not sure what's to be done about that without changing the `Consistbl` API, which feels out-of-scope. Both `Consistbl` and `Import_info` are geared toward expressing that various bits of data either are or are not there, without saying why.

Reply:
Well, the data is of the form:
```ocaml
Cu.Name.t * (Cu.t option * Digest.t) option
```
with 3 possible cases rather than:
```ocaml
Cu.Name.t * Cu.t option * Digest.t option
```
with 4, so switching to that would be a start.

From there it would also be nice to name some cases, so we should replace the inner option with something like:
```ocaml
type sort =
  | Ordinary of Cu.t
  | Parameter
```
Then, by adding a type like:
```ocaml
type nonalias =
  { sort : sort;
    digest : Digest.t }
```
we can have:
```ocaml
val create_nonalias : Cu.Name.t -> nonalias -> t
(** [create_nonalias n { sort; digest }] is [create_ordinary n cu digest] when [sort] is [Ordinary cu] and [create_parameter n digest] when [sort] is [Parameter]. *)

val create : Cu.Name.t -> nonalias option
(** [create n o] is [create_alias n] when [o] is [None] and [create_nonalias n na] when [o] is [Some na]. *)
```
and you can call `create` directly on the stuff coming out of a `Consistbl`.

Reply:
Okay yeah, did basically this (only I used modules so we have an argument for the `Consistbl` functor, and `nonalias` is just a pair type so that it's _exactly_ what `Consistbl` returns).